src/main/java/org/springframework/samples/petclinic/rest/controller/OwnerRestController.java

```diff
@@ -140,6 +140,8 @@ public ResponseEntity<PetDto> addPetToOwner(Integer ownerId, PetFieldsDto petFie
         Owner owner = new Owner();
         owner.setId(ownerId);
         pet.setOwner(owner);
+        PetType petType = this.clinicService.findPetTypeByName(pet.getType().getName());
```

Comment:
For the past few years, I've almost not followed changes in this repo, but it seems like you trying to solve the existing OpenAPI issue by changing the implementation. Maybe it's the right way for "api first" approach, but we added OpenAPI support layer over the existing code. so I'm not sure.
The provided changes look strange, but I guess @arey can decide if they solve the problem.

Reply:
Hey Vitaliy, thanks for the review. 
You're correct, I'm trying to solve the issue in an "api first" approach.
I understand that you can't decide, if the changes solve the issue or not, but I'd be more than happy to address any code smells you've found.